packages/web-pkg/src/components/ItemFilter.vue

```diff
@@ -170,27 +176,20 @@ export default defineComponent({
       emit('selectionChange', unref(selectedItems))
     }
 
-    const sortItems = (items) => {
-      const selectedItemIds = unref(selectedItems).map((i) => getId(i))
-      return items.sort(
-        (a: any, b: any) =>
-          (selectedItemIds.includes(getId(b)) as any) -
-            (selectedItemIds.includes(getId(a)) as any) ||
-          a[props.displayNameAttribute].localeCompare(b[props.displayNameAttribute])
-      )
-    }
-
     const filterTerm = ref()
     const filter = (items, filterTerm) => {
       if (!(filterTerm || '').trim()) {
-        return items
+        return items.map((item) => ({ ...item, isVisible: true }))
       }
-      const usersSearchEngine = new Fuse(items, {
+      const fuse = new Fuse(items, {
         ...defaultFuseOptions,
         keys: props.filterableAttributes as any
       })
 
-      return usersSearchEngine.search(filterTerm).map((r) => r.item)
+      const result = fuse.search(filterTerm)
+      const matchedIds = new Set(result.map((r) => (r.item as any).id))
```

Comment:
Please use the `getId` method to retrieve the item id because the id-attribute can be defined via property.